src/lib_json/json_writer.cpp

```diff
@@ -310,7 +310,12 @@ static String valueToQuotedStringN(const char* value, unsigned length,
     // sequence from occurring.
     default: {
       if (emitUTF8) {
-        result += *c;
+        if (isControlCharacter(*c)) {
```

Comment:
Good catch.
But I think there is no need to add more method.
You can refer these code : https://github.com/nlohmann/json/blob/a861bdb81f1c5ef0adc270952001096b189dbe9e/include/nlohmann/detail/output/serializer.hpp#L406-#L428

`emitUTF` maybe not a good name.

Reply:
From the code you are referring I guess you mean that I should embed that code as part of the current one which is doing the utf8 encoding?

And yeah, `encodeUTF8` would be much better name, but regardless of the encoding, control and special characters should be encoded/escaped.

Reply:
Ok, I've squeezed it a bit, but `utf8ToCodepoint` should not be called when we don't want to do special processing for utf-8, because it offsets the pointer, and this leaved me no choice but to use the ternary if.